scipy/signal/filter_design.py

```diff
@@ -56,6 +56,24 @@ def _is_int_type(x):
         return True
 
 
+def _validate_fs(fs, allow_none=True):
+    """
+    Check if the given sampling frequency is a scalar and raises an exception
+    otherwise. If allow_none is False, also raises an exception for none
+    sampling rates. Returns the sampling frequency as float or none if the
+    input is none.
+    """
+    if not allow_none and fs is None:
+        raise ValueError("Sampling frequency can not be none.")
+    if fs is not None and not np.isscalar(fs):
+        raise ValueError("Sampling frequency fs must be a single scalar.")
+
+    if fs is None:
+        return None
+    else:
+        return float(fs)
```

Comment:
I would simplify the flow here a bit:
```suggestion
    if fs is None:
        if not allow_none:
            raise ValueError("Sampling frequency can not be none.")
    else:  # should be float
        if not np.isscalar(fs):
            raise ValueError("Sampling frequency fs must be a single scalar.")
        fs = float(fs)
    return fs
```

Reply:
Thanks for the suggestion. I incorporated it.